src/versions/versions.service.ts

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { Version } from '.prisma/client';
+
+@Injectable()
+export class VersionsService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(version: string): Promise<Version> {
+    return this.prisma.$transaction(async (prisma) => {
+      return prisma.version.create({
+        data: {
+          version,
+        },
+      });
+    });
+  }
+
+  async getLatest(): Promise<Version | null> {
+    return this.prisma.version.findFirst({
+      orderBy: {
+        id: 'desc',
```

Comment:
Your earlier comment regarding future version pattern changes makes sense for this sort. However, since the creation is restricted to semantic versioning currently, I think it'd be confusing for someone to look at this method for the first time to understand why it's working as intended. And although it's unlikely for requests to be sent out of order, this could break if we send incorrect / older versions.

The version sort seems cleaner and easier to understand. What do you think about using that, and if we need to support a different pattern, we can update the sort / clean up old records (since they'll be moot anyways) when the time comes?

Reply:
Yeah, after thinking about this more, I'm okay with this, fix incoming. I also wanted to double check that we wouldn't have to implement sorting logic for it in the query, but postgres [seems smart enough](https://www.db-fiddle.com/f/nYSA56m1zvrUCtWbfnyTbz/0)